Mothroaches

FluffiestFloof opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Why? Somebody suggested it. Blame Agnus Huey.

This does the following:
- Adds the Mothroaches, which are ghost roles similar to mice. They are just another random critter/pest.
  - They are a bit tankier than mice but have the same damage modifier as Moth.
  - They are three times heavier than a mouse.
  - They can be carried and worn as a hat.
  - They don't have a specific spawner but that can be done if necessary.
  - Reuses the moth squeak sound for screaming.
  - Uses a new moth 'chitter' sound for their speech, alongside a new accent.
- Adds a mothroach hide which is acquired from butchering the "adorable" mothroach.
- Adds a crafting recipe to make a Moth Plushie using the mothroach hide and three cloth.

This **could** be added to the mouse spawner, given its own spawner or kept as a fairly rare encounter.
It's **currently only** in the VentCritters event and the Random Animal Spawner

**Media**
<!-- 
If applicable, add screenshots or videos to showcase your PR. Small fixes/refactors are exempt, but all PRs which make ingame changes 
(adding clothing, items, new features, etc) must include ingame media or the PR will not be merged, in accordance with our PR guidelines.
This makes it much easier for us to merge PRs and find media for progress reports. If you include media in your pull request, we 
may potentially use it in the SS14 progress reports, with clear credit given.

Use screenshot software like Window's built in snipping tool, ShareX, Lightshot, or recording software like ShareX (gif), ScreenToGif, or Open Broadcaster Software (cross platform).
If you're unsure whether your PR will require media, ask a maintainer.

Check one of the boxes below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
The mothroach
![ss (2023-04-28 at 06 43 35)](https://user-images.githubusercontent.com/9823203/235240007-bb39db3b-e649-49c1-953e-77cb41bbf35e.png)

The mothroach but held and worn
![ss (2023-04-28 at 08 54 54)](https://user-images.githubusercontent.com/9823203/235240041-c9883f57-4d72-4056-b9b2-f93dec2a5978.png)

The mothroach life cycle
![ss (2023-04-28 at 08 43 59)](https://user-images.githubusercontent.com/9823203/235240115-a073d200-eaf4-42d8-9bb0-170c6d87f8eb.png)

The squeaky

https://user-images.githubusercontent.com/9823203/235260424-7d758d78-37df-4f02-b8ac-578bb0453d09.mp4


**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged
There are 4 icons for changelog entries: add, remove, tweak, fix. I trust you can figure out the rest.

You can put your name after the :cl: symbol to change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB

Generally, only put things in changelogs that players actually care about. Stuff like "Refactored X system, no changes should be visible" shouldn't be on a changelog.

For writing actual entries, don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers
-->

:cl:
- add: Added mothroaches.

FluffiestFloof committed:
Mothroach init

FluffiestFloof committed:
Removes the petting conflict

Conflicts with the picking up and it's just annoying honestly.

FluffiestFloof committed:
Specific sound for it

FluffiestFloof committed:
Fix incorrect .rsi

FluffiestFloof committed:
Mothroach accent

FluffiestFloof committed:
Refix the incorrect .rsi

Big brain moment

Elijahrane commented on Resources/Prototypes/Nyanotrasen/Entities/Mobs/NPCs/animals.yml:
```diff
@@ -8,4 +8,112 @@
+    - shape:
+        !type:PhysShapeCircle
+        radius: 0.3
+      density: 70
```
these weigh 21 kg?

FluffiestFloof replied:
No clue what conversion it's supposed to be, but whoops, I took exemple on the fox and forgot to re-adapt for slightly bigger than mouse

Elijahrane replied:
multiply density by radius and that's weight in kg

FluffiestFloof replied:
There we go, they are now 3kg which is somewhat more logical for how big they appear on players wearing them.

Elijahrane commented on Resources/Prototypes/Nyanotrasen/Entities/Mobs/NPCs/animals.yml:
```diff
@@ -8,4 +8,112 @@
+      25: Critical
+      50: Dead
+  - type: MovementSpeedModifier
+    baseWalkSpeed : 5
```
no reason not to allow a lower walk speed, since only players use it

FluffiestFloof replied:
I didn't see much reason for a different walk speed since Mouse also has 5 for both walk and sprint but I can probably do same as human with difference

FluffiestFloof committed:
Lowers density of the little bugger

FluffiestFloof committed:
baseWalkSpeed, nyano tags.yml

FluffiestFloof committed:
Mothroach 3kg

FluffiestFloof committed:
Mothroach takes 3x more space as mouse

Just to match the 3x heavier than mouse

FluffiestFloof committed:
Grindable, SmallMob, Moth damage modifier

Make the bugger grindable into blood and nutriment
Use the normal Moth damage modifier set, meaning they will have the same weakness/resistance
Lowers the asphyxiation rate to match mouse better

FluffiestFloof committed:
Delete VentCritters.cs

FluffiestFloof committed:
Merge branch 'Nyanotrasen:master' into mothroach

FluffiestFloof committed:
Update mothroachs to latest changes
